### ISSUE TITLE
Wacom touch rotation\left handed mode not working

### ISSUE THREAD

**Issue by khalid151** [swaywm/sway#4068]
Rotating tablet (Wacom Intuos Pro, PTH-660) with `swaymsg input <identifier> left_handed enabled` does not rotate touch input of the tablet along with the pad (the pad is rotated as expected).

I've tried enabling that option on _Wacom Intuos Pro M Pen_, _Pad_ and _Finger_ but still getting the same issue.

```
sway version 1.0-rc1-243-g35ddd9aa (Apr 18 2019, branch 'master')
```

**Comment by emersion** [comment]
I wonder if this is a libinput bug.

**Comment by khalid151** [comment]
Would using libinput directly to configure the device help? If so, I might try that when I have time as soon as possible.

This has been an issue for a while now and I thought it was a sway thing.

**Comment by RedSoxFan** [comment]
When you run `swaymsg -rt get_inputs`, it does show `"left_handed": "enabled"` in the `libinput` section for the device, right? If not, the libinput call to set left handed mode is returning an error and you can look at the debug log (start sway with `sway -d >sway.log 2>&1`).

You can also try `sudo libinput debug-events --enable-left-handed` and see what gets outputted for coordinates when using the tablet

**Comment by khalid151** [comment]
I don't have access to the tablet for the moment but I've tried testing with my laptop's touch pad since the touch input of the wacom tablet (Intuos) is basically a giant touch pad.

Running `swaymsg -rt get_inputs` does show `"left_handed": "enabled"` for the touch pad but it's still not rotated.

Could this mean libinput handles such devices differently?

**Comment by RedSoxFan** [comment]
> Could this mean libinput handles such devices differently?

Yes. https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga9bbdef04c07804ce7c121133c04d4edf

**Comment by khalid151** [comment]
Is there another way to achieve that? Like a transformation matrix maybe?

**Comment by RedSoxFan** [comment]
> Is there another way to achieve that? Like a transformation matrix maybe?

If you mean to rotate the touchpad, there is https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga136b2f7753235c5d2a9fb2762a040eef which sway doesn't currently support, but seems trivial to implement.

**Comment by khalid151** [comment]
Okay, thanks!
I've also seen [`libinput_device_config_calibration_set_matrix()`](https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga09a798f58cc601edd2797780096e9804).
I will test both of them when I have time and will report back.

**Comment by khalid151** [comment]
Well, both of `libinput_device_config_calibration_has_matrix()` and `libinput_device_config_rotation_is_available()` returns 0. Seems like there's no supported way by libinput to rotate touchpad (and in this case, wacom touch\finger device).

**Comment by khalid151** [comment]
Fixed in [`libinput 1.13.900`](https://gitlab.freedesktop.org/libinput/libinput/merge_requests/217)
